packages/react-components/src/utils/setupConfiguration.ts

```diff
@@ -343,6 +364,62 @@ function getSeverityFilterValue(selectedSeverities: Severity[], allSeverities: S
     return severitiesFilter;
 }
 
+function convertSourceTypeFilterToArray(
+    sourceTypesFilter: string[],
+    allSourceTypeCategories: SourceTypeCategory[]
+): SourceType[] {
+    if (sourceTypesFilter.length === 0) {
+        return [...allSourceTypeCategories.flatMap((category) => category.types)];
+    }
+    const sourceTypes: SourceType[] = [];
+    sourceTypesFilter.forEach((sourceTypeValue) => {
+        const foundSourceTypeCategoryMatch = allSourceTypeCategories.find(
+            (sourceTypeCategory) => sourceTypeCategory.value === sourceTypeValue
+        );
+        if (foundSourceTypeCategoryMatch) {
+            sourceTypes.push(...foundSourceTypeCategoryMatch.types);
+        } else {
+            const foundSourceTypeMatch = allSourceTypeCategories
+                .flatMap((category) => category.types)
+                .find((sourceType) => sourceType.value === sourceTypeValue);
+            if (foundSourceTypeMatch) {
+                sourceTypes.push(foundSourceTypeMatch);
+            }
+        }
+    });
+    return sourceTypes;
+}
```

Comment:
This is difficult to read. Some contextual spacing would help a bit. Also, less indentation. Can lose the else block.